kafka/adminapi.go

```diff
@@ -76,6 +76,16 @@ import "C"
 type AdminClient struct {
 	handle    *handle
 	isDerived bool // Derived from existing client handle
+
+	isClosed bool // to check if Admin Client is closed or not.
+}
+
+// verifyAdminClient verifies if AdminClient can be used
+func (a *AdminClient) verifyAdminClient() bool {
+	if a == nil || a.isClosed {
```

Comment:
Can `a == nil` even happen? Sounds like a usage error:
```go
a := NewAdminClient(..)
... other stuff..

a = nil

a.DoThing()  // invalid, not something we need to support
```

Reply:
I think what someone actually did was this -  
```
a, err := NewAdminClient(..)
// err is not nil, a is nil, now if I use a without checking err, it is a panic.
a.DoThing()
```

Reply:
@milindl I think it's impossible to have a=nil when NewAdminClient has been called.

The only way I see this is usage error by assigning a = nil.

Reply:
There are a couple of places in NewAdminClient where we return `nil, err` from, all those would indeed cause this.

Reply:
Sorry, mistook it thought err is also nil. Agreed.

Reply:
Right, not checking the err return is a programming error and a huge golang antipattern, so not something we need to explicitly handle.

Reply:
Moving ahead without nil check. 

cc: @edenhill @milindl